### PR TITLE
[Backport] HGCAL trigger updates - 1

### PR DESCRIPTION
**Pull request by jbsauvan**
This is a backport of #22387, except for the commit that adds the HGCAL TPG in standard sequences, which will be added in a second PR.

### CODE REVIEW

**Comment by rekovic** [comment]
please test

**Comment by kpedro88** [comment]
backport of #22387

**Comment by kpedro88** [comment]
@smuzaffar do we need to make a corresponding backport of https://github.com/cms-sw/cmsdist/pull/3785 for 93X branches?

**Comment by rekovic** [comment]
@jbsauvan 
Will the part-2 PR that you are planning to make, and which configures hgcTPs and include them in the sequence be orthogonal this PR?  I hope so. Can you please provide it so we can test with and without it. Thanks.

**Comment by jbsauvan** [comment]
Hi @rekovic 
The commit that switched the HGCAL TPG on  ([here](https://github.com/PFCal-dev/cmssw/commit/41dffc2b458507b3f4ab2cb034fc5b1301dbbe2b)) also included other things:
1) Changes in the TPG configuration: change to a new clustering that was not implemented before
2) Updates in classes_def.xml
3) Change in the event content, adding in particular towers, which were not implemented before

But I could apply 1), 2), and 3) here, and just uncomment the HGCAL TPG in SimL1Emulator_cff.py in the second PR, in order to de-correlate the 2 PRs.

**Comment by jbsauvan** [comment]
Hi @rekovic 
I have updated this PR with the changes 1), 2), 3) listed above, and created the second PR with only the HGCAL TPG switch (#22487)

File: L1Trigger/Configuration/python/L1Trigger_EventContent_cff.py [hunk 1]
```diff
@@ -117,7 +117,10 @@ def _appendStage2Digis(obj):
 # adding HGCal L1 trigger digis
 def _appendHGCalDigis(obj):
     l1HGCalDigis = [
-        'keep *_hgcalTriggerPrimitiveDigiProducer__*',
+        'keep *_hgcalTriggerPrimitiveDigiProducer_calibratedTriggerCells_*',
```
Comment (rekovic) [inline L1Trigger/Configuration/python/L1Trigger_EventContent_cff.py]: @jbsauvan 
What is the reason for further specialization of `hgcalTriggerPrimitiveDigiProducer`  data too keep in these lines?  What are we dropping?

Comment (jbsauvan) [reply L1Trigger/Configuration/python/L1Trigger_EventContent_cff.py]: We are dropping:
- The collection of `DataFormats/L1THGCal/HGCFETriggerDigi`, which are basically vectors of bool encoding the trigger cell raw charge. `HGCalTriggerCell` objects contain more information (e.g. calibrated energies) and are easier to use.
- The collection of `DataFormats/L1THGCal/HGCalTowerMap`, which are objects used to compute the towers

**Comment by rekovic** [comment]
please test

File: L1Trigger/Configuration/python/L1Trigger_EventContent_cff.py (hunk 1 above)
Comment (rekovic) [reply L1Trigger/Configuration/python/L1Trigger_EventContent_cff.py]: *  >The collection of DataFormats/L1THGCal/HGCFETriggerDigi, which are basically vectors of bool encoding the trigger cell raw charge. HGCalTriggerCell objects contain more information (e.g. calibrated energies) and are easier to use.

OK, there we don't lose any info, I understand. 

 * > The collection of DataFormats/L1THGCal/HGCalTowerMap, which are objects used to compute the towers

Here there might be some info lost that we might need in the future.  Could not tell for sure
what L1T algo development could need. 

I rather would not drop anything from HGC digis, unless completely necessary.

Comment (jbsauvan) [reply L1Trigger/Configuration/python/L1Trigger_EventContent_cff.py]: The HGCAL TPGs are only the 3D clusters and the towers. All other objects will not be available to the L1T system.
But ok, I can add the `HGCalTowerMap` objects, which give the tower energies in each layer.

Comment (kpedro88) [reply L1Trigger/Configuration/python/L1Trigger_EventContent_cff.py]: Remember to propagate the update to #22387 as well

Comment (jbsauvan) [reply L1Trigger/Configuration/python/L1Trigger_EventContent_cff.py]: Thanks for the reminder; Yes I will.

**Comment by kpedro88** [comment]
I made https://github.com/cms-sw/cmsdist/pull/3815 to backport the external update.

**Comment by kpedro88** [comment]
please test

**Comment by smuzaffar** [comment]
please test with cms-sw/cmsdist#3815

**Comment by smuzaffar** [comment]
please test with cms-sw/cmsdist#3815

**Comment by rekovic** [comment]
+1

**Comment by kpedro88** [comment]
+1

**Comment by rekovic** [comment]
@fabiocos Can we please merge this one.

**Comment by kpedro88** [comment]
merge

**Pull request merged**